### PR TITLE
Report percentage attributes as double

### LINKED ISSUE #3: [lvmdbusd] Percentage Values
Currently the percentage values such as MdaPercent + DataPercent wouldn't be exposed correctly. These are always zero.

Is there any reason why the percentages aren't reported as double, if no i'd like to change them.

This has been reported over dbus.
```
'/com/redhat/lvmdbus1/Lv/54': {'com.redhat.lvmdbus1.Lv': {},
                                'com.redhat.lvmdbus1.LvCommon': {'Active': True,
                                                                 'AllocationPolicy': ('i',
                                                                                      'inherited'),
                                                                 'Attr': 'Vwi-a-tz--',
                                                                 'CopyPercent': 0,
                                                                 'DataPercent': 0,
                                                                 'Devices': [],
                                                                 'FixedMinor': False,
                                                                 'Health': ('-',
                                                                            'Unspecified'),
                                                                 'HiddenLvs': [],
                                                                 'IsThinPool': False,
                                                                 'IsThinVolume': True,
                                                                 'MetaDataPercent': 0,
                                                                 'MetaDataSizeBytes': 0,
                                                                 'MovePv': '/',
                                                                 'Name': 'thin_lv',
                                                                 'OriginLv': '/',
                                                                 'Path': '/dev/tonobo-top-vg/thin_lv',
                                                                 'Permissions': ('w',
                                                                                 'writable'),
                                                                 'PoolLv': '/com/redhat/lvmdbus1/ThinPool/0',
                                                                 'Roles': ['public'],
                                                                 'SegType': ['thin'],
                                                                 'SizeBytes': 3221225472,
                                                                 'SkipActivation': False,
                                                                 'SnapPercent': 0,
                                                                 'State': ('a',
                                                                           'active'),
                                                                 'SyncPercent': 0,
                                                                 'Tags': [],
                                                                 'TargetType': ('t',
                                                                                'thin'),
                                                                 'Uuid': 'O0LeqP-mnW4-27dn-fi76-utXN-oVng-AFcLC8',
                                                                 'Vg': '/com/redhat/lvmdbus1/Vg/0',
                                                                 'VolumeType': ('V',
                                                                                'thin '
                                                                                'Volume'),
                                                                 'ZeroBlocks': True}},
```

This is what lvs has been returned.
```
timmy@tonobo-top  >>> sudo lvs -o name,data_percent,metadata_lv,metadata_percent | grep thin_lv
  thin_lv   6.51  
```

### PR DESCRIPTION
**Pull request by tonobo**
Fixes #3

### CODE REVIEW

File: daemons/lvmdbusd/lv.py [hunk 1]
```diff
@@ -232,7 +232,6 @@ def creation_signature(self):
 @utils.dbus_property(LV_COMMON_INTERFACE, 'Attr', 's')
 @utils.dbus_property(LV_COMMON_INTERFACE, 'DataPercent', 'u')
 @utils.dbus_property(LV_COMMON_INTERFACE, 'SnapPercent', 'u')
-@utils.dbus_property(LV_COMMON_INTERFACE, 'DataPercent', 'u')
```
Comment (tasleson) [inline daemons/lvmdbusd/lv.py]: The third parameter is the dbus type specifier `u` is a **Unsigned 32-bit integer**.  This is part of the public API specification, so right or wrong it's part of the API.  We can certainly add new properties which are a double, but we have clients which now expect that this is an unsigned 32 bit integer.

**Comment by tasleson** [comment]
We can certainly fix the issue where we are not getting a value, but like I mentioned in my line comment, we really cannot muck with the public facing API.

File: daemons/lvmdbusd/lv.py (hunk 1 above)
Comment (tonobo) [reply daemons/lvmdbusd/lv.py]: Yes I know, possibly its a bad idea to break the API :smile:. But the line your referencing is a duplicate and could be removed.

**Comment by tonobo** [comment]
Possibly we could introduce a separated option or is there any way to get the amount of used bytes?

File: daemons/lvmdbusd/lv.py (hunk 1 above)
Comment (tasleson) [reply daemons/lvmdbusd/lv.py]: Like I mentioned, we can certainly fix bugs, and having a duplicate line here is indeed a bug.

**Comment by tasleson** [comment]
> Possibly we could introduce a separated option or is there any way to get the amount of used bytes?

Best would be to introduce a new property if it's something that is needed and useful.  As always pull requests welcomed :-)

**Comment by tonobo** [comment]
I'll take a look at it tomorrow. In general I prefer to get exact values instead of getting inaccurate percentages.

File: daemons/lvmdbusd/lv.py (hunk 1 above)
Comment (tonobo) [reply daemons/lvmdbusd/lv.py]: I'll rework the necessary code, so you possibly just need squash on merge.

**Comment by tonobo** [comment]
For getting accurate usage statistics I only need to expose the mapped_sectors from `dm_status_thin`?

**Comment by tasleson** [comment]
@tonobo I cherry picked and committed the duplicate line correction upstream, thank you!

**Comment by tonobo** [comment]
Nice :+1:

**Pull request closed without merging**